src/core/mac/mac_frame.cpp

```diff
@@ -1333,6 +1337,56 @@ Error TxFrame::GenerateEnhAck(const RxFrame &aFrame, bool aIsFramePending, const
 }
 #endif // OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
 
+#if OPENTHREAD_CONFIG_PSA_CRYPTO_ENABLE
+Error RxFrame::ProcessReceiveAesCcm(const ExtAddress &aExtAddress, otMacKeyRef aMacKeyRef)
+{
+#if OPENTHREAD_RADIO
+    OT_UNUSED_VARIABLE(aExtAddress);
+    OT_UNUSED_VARIABLE(aMacKey);
+
+    return kErrorNone;
+#else
+    Error          error        = kErrorSecurity;
+    uint32_t       frameCounter = 0;
+    uint8_t        securityLevel;
+    uint8_t        nonce[Crypto::AesCcm::kNonceSize];
+    uint8_t        tag[kMaxMicSize];
+    uint8_t        tagLength;
+    Crypto::AesCcm aesCcm;
+
+    VerifyOrExit(GetSecurityEnabled(), error = kErrorNone);
+
+    SuccessOrExit(GetSecurityLevel(securityLevel));
+    SuccessOrExit(GetFrameCounter(frameCounter));
+
+    Crypto::AesCcm::GenerateNonce(aExtAddress, frameCounter, securityLevel, nonce);
+
+    aesCcm.SetKey(aMacKeyRef);
```

Comment:
I think having a common `KeyRef` type (and  `otPlatKeyRef`) which can contain either PSA ref or a full key (pointer to key bytes) and then using as input type to this method will help simplify here as well.

This method is almost a copy of the other version under `#else`. The difference between the two seems to be on how the key is being passed in as the input to the method.

I think we want to avoid repeating large chunks of code and similar implementations of same method, since this can make it hard to maintain in future. For example, if there is a fix or a change in this method in future, we need to remember to update both versions. It's better if we can have a single version that can be used under both configs.

Reply:
Yes, I do agree to this point. I can make these changes to the PR. We just need to decide if we want to support both mbedTLS and PSA, or move to PSA entirely.